Sources/MapboxMaps/Viewport/ViewportStatusObserver.swift

```diff
@@ -0,0 +1,19 @@
+public protocol ViewportStatusObserver: AnyObject {
+    func viewportStatusDidChange(from fromStatus: ViewportStatus,
+                                 to toStatus: ViewportStatus,
+                                 reason: ViewportStatusChangeReason)
+}
+
+public struct ViewportStatusChangeReason: RawRepresentable, Hashable {
+    public typealias RawValue = String
+
+    public var rawValue: String
+
+    public init(rawValue: String) {
+        self.rawValue = rawValue
+    }
+
+    public static let programmatic = ViewportStatusChangeReason(rawValue: "PROGRAMMATIC")
+
+    public static let userInteraction = ViewportStatusChangeReason(rawValue: "USER_INTERACTION")
+}
```

Comment:
@pengdev here's how I modeled the change reason

Reply:
this is a plain string on Android, we could consider make it a separate type to make it extendable in the future, will add to my todo list before RC.